docs/learn/security-foundations/security-testing-workflow.md

```diff
@@ -0,0 +1,70 @@
+- Effective tools also provide **remediation guidance**, offering developers clear, actionable steps to resolve problems rather than just pointing them out.
+
+Research has shown that customizable rules with active remediation guidance greatly improve the developer experience and result in higher fix rates [1-4].
+
+## Integration into developer workflows
+
+![Testing Diagram](./assets/workflow.png)
+
+### Local development
+
+The most powerful place to integrate security testing is directly into the developer’s daily environment.
+
+- **IDE Integration:** Providing real-time feedback inside the IDE allows developers to see issues as they type and receive inline fix suggestions. Research has shown that this approach can increase fix rates dramatically, in some cases up to 98% [1-3].
+- **CLI and Manual Scans:** Developers who prefer command-line workflows can run deeper, on-demand scans before committing their code.
+- **Pre-commit Hooks:** Serving as the last line of defense before code enters source control, pre-commit hooks can be configured to block commits only when rules are proven to be highly accurate for a given codebase.
+- **MCP Server Integration:** As AI-assisted coding becomes more common, integrating security checks directly into model-context protocol (MCP) servers ensures that LLM-generated code is automatically reviewed with context-aware security suggestions.
+
+### CI/CD Pipeline
+
+Security checks in the CI/CD pipeline provide additional layers of assurance without disrupting the developer’s inner loop.
+
+- **Pre-receive Hooks:** Depending on platform support, these serve as a final gate before code reaches the main branch.
+- **SCM Integration Features:** Diff-aware scanning helps security tools focus on newly changed code, while automated comments on pull requests give developers immediate feedback. Status checks can also block merges for specific high-confidence rules.
+- **Build Integration:** Teams can choose between incremental scans for speed or full scans for comprehensive coverage, depending on the stage of the build.
+
+### Reporting and Tracking
+
+Security findings need to be tracked and communicated effectively to avoid becoming noise.
+
+- **Output Formats:** Supporting common formats such as SARIF, JSON, CSV, and SBOM ensures compatibility with other systems.
+- **Dashboard Requirements:** Dashboards should go beyond raw counts of vulnerabilities, offering trend analysis, mean time to remediation (MTTR), vulnerability density metrics, and adoption rates among developers. They should also make it easy to triage findings, and update rules to increase confidence.
+- **Team Communication:** Automated workflows ensures the right findings are prioritized and fixed within the required timeframes. The tool should integrate with the ticketing system, like Jira or Linear. It could also be enabled to send notifications to Slack or Teams, or even used for triggering webhooks for custom integrations.
```

Comment:
```suggestion
- **Team Communication:** Automated workflows ensures the right findings are prioritized and fixed within the required time frames. The tool should integrate with the ticketing system, like Jira or Linear. It could also be enabled to send notifications to Slack or Teams, or even used for triggering webhooks for custom integrations.
```